src/cargo/util/network.rs

```diff
@@ -1,4 +1,30 @@
-use util::{CargoResult, Config, errors};
+use util::Config;
+use util::errors::{CargoError, CargoErrorKind, CargoResult};
+
+use git2;
+
+fn maybe_spurious(err: &CargoError) -> bool {
+    match err.kind() {
```

Comment:
This would need a trick like the one in lib.rs that walks the chain to print. There, I could convince myself that the `transmute`-to-static trick is safe because the error was moved into the method and consumed entirely during printing. Here, we start with a `&CargoError`. I don't have the confidence to say that transmuting the chain to static and downcasting here is similarly ok, just because I'm still too new to Rust. It seems like it might be ok? Maybe? I can at least put it in, and people wiser than I can review, and then just pull it back out if it's no good.